src/marqo/core/unstructured_vespa_index/unstructured_vespa_schema.py

```diff
@@ -172,81 +164,172 @@ def _generate_unstructured_schema(cls, marqo_index: UnstructuredMarqoIndex) -> s
-                    summary {cls._INT_FIELDS} type map<string, long> {{}}
-                    summary {cls._FLOAT_FIELDS} type map<string, double> {{}}
-                    summary {cls._CHUNKS} type array<string> {{}}
+        # Add summaries
+        schema += textwrap.dedent(
+            f"""
+                document-summary {self._SUMMARY_ALL_NON_VECTOR} {{
+                    summary {self._FIELD_ID} type string {{}}
+                    summary {self._STRINGS} type array<string> {{}}
+                    summary {self._LONGS_STRINGS_FIELDS} type map<string, string> {{}}
+                    summary {self._SHORT_STRINGS_FIELDS} type map<string, string> {{}}
+                    summary {self._STRING_ARRAY} type array<string> {{}}
+                    summary {self._BOOL_FIELDS} type map<string, byte> {{}}
+                    summary {self._INT_FIELDS} type map<string, long> {{}}
+                    summary {self._FLOAT_FIELDS} type map<string, double> {{}}
+                    summary {self._CHUNKS} type array<string> {{}}
                 }}
 
-                document-summary {cls._SUMMARY_ALL_VECTOR} {{
-                    summary {cls._FIELD_ID} type string {{}}
-                    summary {cls._STRINGS} type array<string> {{}}
-                    summary {cls._LONGS_STRINGS_FIELDS} type map<string, string> {{}}
-                    summary {cls._SHORT_STRINGS_FIELDS} type map<string, string> {{}}
-                    summary {cls._STRING_ARRAY} type array<string> {{}}
-                    summary {cls._BOOL_FIELDS} type map<string, byte> {{}}
-                    summary {cls._INT_FIELDS} type map<string, long> {{}}
-                    summary {cls._FLOAT_FIELDS} type map<string, double> {{}}
-                    summary {cls._CHUNKS} type array<string> {{}}
-                    summary {cls._EMBEDDINGS} type tensor<float>(p{{}}, x[{dimension}]) {{}}
+                document-summary {self._SUMMARY_ALL_VECTOR} {{
+                    summary {self._FIELD_ID} type string {{}}
+                    summary {self._STRINGS} type array<string> {{}}
+                    summary {self._LONGS_STRINGS_FIELDS} type map<string, string> {{}}
+                    summary {self._SHORT_STRINGS_FIELDS} type map<string, string> {{}}
+                    summary {self._STRING_ARRAY} type array<string> {{}}
+                    summary {self._BOOL_FIELDS} type map<string, byte> {{}}
+                    summary {self._INT_FIELDS} type map<string, long> {{}}
+                    summary {self._FLOAT_FIELDS} type map<string, double> {{}}
+                    summary {self._CHUNKS} type array<string> {{}}
+                    summary {self._EMBEDDINGS} type tensor<float>(p{{}}, x[{dimension}]) {{}}
                 }}
             }}
             """
         )
+
+        return schema
+
+    def _generate_rank_profiles(self, marqo_index: UnstructuredMarqoIndex):
+        rank_profiles: List[str] = list()
```

Comment:
This implementation seems inconsistent with the rest of this class. Since the schema is the same for all indexes, we have almost hard-coded it (with values interpolated) which is more readable. I think it's best we keep the class consistent.

Reply:
I'm following the structured_vespa_schema pattern for this one. I found it easier to modularize when we add things line by line. Do you want to follow that approach?

Reply:
Nevermind, fixed to match the rest of the class.